README.md

```diff
@@ -7,59 +7,45 @@
 
-## Philosophy
+By placing our build logic inside of images, and by combining the images into multiple steps, we can keep our runtime environment close to our build environment (same JDK, same Tomcat JARs) without requiring build tools to be deployed to production.
 
-1. Simplify the process of application source + builder image -> usable image for most use cases (the
-   80%)
-1. Define and implement a workflow for incremental builds that eventually uses only Docker
-   primitives
-1. Develop tooling that can assist in verifying that two different builder images result in the same
-   `docker run` outcome for the same input
-1. Use native Docker primitives to accomplish this - map out useful improvements to Docker that
-   benefit all image builders
-1. Allow users to build new application images in a controlled, secure, restricted environment
+Learn more about [building your own images](#getting-started).
 
-## Codified Image Construction Patterns
 
-### Image flexibility   
-S2I scripts can be written to inject application code into almost any existing Docker image, taking advantage of the existing image ecosystem.
+## Goals
 
-### Speed   
-With S2I, the assemble process can perform a large number of complex operations without creating a new layer at each step, resulting in a fast process. In addition, S2I scripts can be written to re-use artifacts stored in a previous version of the application image, rather than having to download or build them each time the build is run.
+### Reproducibility
+Allow build environments to be tightly versioned by encapsulating them within a Docker image and defining a simple interface (injected source code) for callers.
 
-### Patchability  
-S2I allows you to rebuild the application in a consistent way if an underlying image needs a patch due to a security issue.
+### Flexibility   
+The scripts that process the application source code can be injected into the builder image, allowing an author to adapt existing images.
 
-### Operational security  
-Building an arbitrary Dockerfile exposes the host system to root privilege escalation. This can be exploited by a malicious user because the entire Docker build process is run as a user with Docker privileges. S2I restricts the operations that can be performed as a root user and can run user
-supplied build scripts as a non-root user.  Furthermore, operations teams can supply whitelisted builder images which will not perform
-arbitrary actions such as installing random packages or downloading content from untrusted sources.
+### Speed   
+Instead of building multiple layers in a single Dockerfile, S2I encourages (and benefits from) authors to set up an application in a single image layer. This is both more powerful and faster than classic multi-layer builds.
 
-### Ecosystem
-S2I encourages a shared ecosystem of images where you can leverage best practices for your applications and build processes.
```

Comment:
We need to say something about s2i working with pretty much any existing image and you  can either add scripts to the image, or provide the scripts externally and use the image as is.